libs/posthog_client_lite/src/lib.rs

```diff
@@ -0,0 +1,616 @@
+//! A lite version of the PostHog client that only supports local evaluation of feature flags.
+
+use std::{collections::HashMap, sync::Arc};
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use sha2::Digest;
+
+#[derive(Debug, thiserror::Error)]
+pub enum PostHogEvaluationError {
+    /// The feature flag is not available, for example, because the local evaluation data is not populated yet.
+    #[error("Feature flag not available: {0}")]
+    NotAvailable(&'static str),
+    #[error("No condition group is matched")]
+    NoConditionGroupMatched,
+    /// Real errors, e.g., the rollout percentage does not add up to 100.
+    #[error("Failed to evaluate feature flag: {0}")]
+    Internal(&'static str),
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationResponse {
+    #[allow(dead_code)]
+    flags: Vec<PostHogLocalEvaluationFlag>,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlag {
+    key: String,
+    filters: PostHogLocalEvaluationFlagFilters,
+    active: bool,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilters {
```

Comment:
nit: technically, we already have `posthog` in lib name, probably it's not worth it to add `PostHog` to every class name. E.g. for the client I'm OK with `PostHogClient`, because there are too many `Client`s already. But for those structs the name is already pretty long and unique.